processor/filterprocessor/filter_processor_traces_test.go

```diff
@@ -0,0 +1,181 @@
+	resourceAttributes map[string]pdata.AttributeValue
+	tags               map[string]pdata.AttributeValue
+}
+
+// All the data we need to define a test
+type traceTest struct {
+	name              string
+	inc               *filterconfig.MatchProperties
+	exc               *filterconfig.MatchProperties
+	inTraces          pdata.Traces
+	allTracesFiltered bool
+	spanCountExpected int // The number of spans that should be left after all filtering
+}
+
+var (
+	redisTraces = []testTrace{
+		{
+			spanName:       "test!",
+			libraryName:    "otel",
+			libraryVersion: "11",
+			resourceAttributes: map[string]pdata.AttributeValue{
+				"service.name": pdata.NewAttributeValueString("test_service"),
+			},
+			tags: map[string]pdata.AttributeValue{
+				"db.type": pdata.NewAttributeValueString("redis"),
+			},
+		},
+	}
+
+	nameTraces = []testTrace{
+		{
+			spanName:       "test!",
+			libraryName:    "otel",
+			libraryVersion: "11",
+			resourceAttributes: map[string]pdata.AttributeValue{
+				"service.name": pdata.NewAttributeValueString("keep"),
+			},
+		},
+		{
+			spanName:       "test!",
+			libraryName:    "otel",
+			libraryVersion: "11",
+			resourceAttributes: map[string]pdata.AttributeValue{
+				"service.name": pdata.NewAttributeValueString("dont_keep"),
+			},
+		},
+		{
+			spanName:       "test!",
+			libraryName:    "otel",
+			libraryVersion: "11",
+			resourceAttributes: map[string]pdata.AttributeValue{
+				"service.name": pdata.NewAttributeValueString("keep"),
+			},
+		},
+	}
+
+	serviceNameMatchProperties = &filterconfig.MatchProperties{
+		Config:   filterset.Config{MatchType: filterset.Strict},
+		Services: []string{"keep"},
+	}
+
+	redisMatchProperties = &filterconfig.MatchProperties{
+		Attributes: []filterconfig.Attribute{
+			{Key: "db.type", Value: "redis"},
+		},
+	}
+
+	standardTraceTests = []traceTest{
+		{
+			name:              "filterRedis",
+			exc:               redisMatchProperties,
+			inTraces:          generateTraces(redisTraces),
+			allTracesFiltered: true,
+		},
+		{
+			name:              "keepRedis",
+			inc:               redisMatchProperties,
+			inTraces:          generateTraces(redisTraces),
+			spanCountExpected: 1,
+		},
+		{
+			name:              "keepServiceName",
+			inc:               serviceNameMatchProperties,
+			inTraces:          generateTraces(nameTraces),
+			spanCountExpected: 2,
+		},
+	}
+)
+
+func TestFilterTraceProcessor(t *testing.T) {
+	for _, test := range standardTraceTests {
+		t.Run(test.name, func(t *testing.T) {
+			ctx := context.Background()
+			next := new(consumertest.TracesSink)
+			cfg := &Config{
+				ProcessorSettings: config.NewProcessorSettings(config.NewComponentID(typeStr)),
+				Spans: SpanFilters{
+					Include: test.inc,
+					Exclude: test.exc,
+				},
+			}
+			factory := NewFactory()
+			fmp, err := factory.CreateTracesProcessor(
+				ctx,
+				componenttest.NewNopProcessorCreateSettings(),
+				cfg,
+				next,
+			)
+			assert.NotNil(t, fmp)
```

Comment:
At this point, you should probably use `require` instead of `assert`, given that you are still setting up the scene for the test.

Reply:
good call, fyi this is copy and paste carry over, so this type of stuff is all over the place